app/jenkins-for-jira-ui/src/components/JenkinsSetup/JenkinsSetup.tsx

```diff
@@ -219,37 +241,67 @@ const JenkinsSetup = (): JSX.Element => {
 		fetchData();
 	}, [uuid, getServer]);
 
-	const handleCopyToClipboard = (copyRef: React.RefObject<HTMLDivElement>) => {
-		if (copyRef.current) {
-			const range = document.createRange();
-			range.selectNode(copyRef.current);
-
-			const selection = window.getSelection();
-			if (selection) {
-				selection.removeAllRanges();
-				selection.addRange(range);
+	const handleCopyToClipboard =
+		async (copyRef: React.RefObject<HTMLDivElement>, elementName?: string) => {
+			await analyticsClient.sendAnalytics(
+				AnalyticsEventTypes.UiEvent,
+				AnalyticsUiEventsEnum.CopiedToClipboardName,
+				{
+					source: AnalyticsScreenEventsEnum.JenkinsSetupScreenName,
+					buttonClicked: elementName || 'Copy button'
+				}
+			);
+
+			if (copyRef.current) {
+				const range = document.createRange();
+				range.selectNode(copyRef.current);
+
+				const selection = window.getSelection();
+				if (selection) {
+					selection.removeAllRanges();
+					selection.addRange(range);
+				}
+
+				try {
+					document.execCommand('copy');
+				} catch (err) {
+					console.error('Copy to clipboard failed:', err);
+				}
+
+				if (selection) {
+					selection.removeAllRanges();
+				}
 			}
+		};
 
-			try {
-				document.execCommand('copy');
-			} catch (err) {
-				console.error('Copy to clipboard failed:', err);
-			}
+	const handleMyJenkinsAdminClick = async (e: React.MouseEvent) => {
+		e.preventDefault();
 
-			if (selection) {
-				selection.removeAllRanges();
+		await analyticsClient.sendAnalytics(
+			AnalyticsEventTypes.UiEvent,
+			AnalyticsUiEventsEnum.MyJenkinsAdminTabName,
+			{
+				source: AnalyticsScreenEventsEnum.JenkinsSetupScreenName
 			}
-		}
-	};
+		);
 
-	const handleMyJenkinsAdminClick = (e: React.MouseEvent) => {
-		e.preventDefault();
 		setShowMyJenkinsAdmin(true);
 		setShowIAmTheJenkinsAdmin(false);
 	};
 
-	const handleIAmTheJenkinsAdminClick = (e: React.MouseEvent) => {
+	const handleIAmTheJenkinsAdminClick = async (e: React.MouseEvent) => {
 		e.preventDefault();
+
+		await analyticsClient.sendAnalytics(
```

Comment:
Same as above with try/catch or move to final step in the function, alternatively you could catch the errors in analyticsClient.sendAnalytics itself and just carry on